### ISSUE TITLE
Acrylic Broken

### ISSUE THREAD

**Issue by Mukul1127** [microsoft/terminal#12356]
### Windows Terminal version

1.13.10336.0

### Windows build number

Microsoft Windows [Version 10.0.19044.1503]

### Steps to reproduce

1. Install The Latest Version Of Windows Terminal Preview On Windows 10.
2. Boom, Title bar Acrylic Broken

### Expected Behavior

Acrylic On Title Bar Works

### Actual Behavior

Acrylic Title Bar Very White

**Comment by zadjii-msft** [comment]
Could you share your [settings.json file](https://github.com/microsoft/terminal/wiki/Frequently-Asked-Questions-(FAQ)#where-can-i-find-the-settings-file)?

Does acrylic work anywhere else in the OS?

**Comment by Snowknight26** [comment]
In case it's not a coincidence, it could be an [AMD driver issue](https://old.reddit.com/r/Amd/comments/s1o19w/radeon_software_adrenalin_2211_release_notes/hsb8jjt/?context=10000).

**Comment by pranavAbe** [comment]
@zadjii-msft I've been facing the same issue since the v1.13.10336 release of Terminal Preview yesterday. The acrylic has stopped working inside Terminal Preview, and works fine everywhere else in the system. 

I have the `"useAcrylic"` set to `True` under `"defaults"` of `"profiles"` in the settings.json, and it used to work just fine prior to this update. 

Attaching my settings.json here (attached as a TEXT file since GitHub doesn't seem to support JSON file uploads: [settings.txt](https://github.com/microsoft/terminal/files/8001655/settings.txt)

**Comment by pranavAbe** [comment]
@zadjii-msft This issue resolved itself after a restart. Not sure how, but I don't have the problem anymore.

**Comment by zadjii-msft** [comment]
Huh. Weird. Well, hopefully it was just a one off glitch 😅

**Comment by Mukul1127** [comment]
> @zadjii-msft I've been facing the same issue since the v1.13.10336 release of Terminal Preview yesterday. The acrylic has stopped working inside Terminal Preview, and works fine everywhere else in the system.
> 
> I have the `"useAcrylic"` set to `True` under `"defaults"` of `"profiles"` in the settings.json, and it used to work just fine prior to this update.
> 
> Attaching my settings.json here (_attached as a TEXT file since GitHub doesn't seem to support JSON file uploads_): [settings.txt](https://github.com/microsoft/terminal/files/8001655/settings.txt)

That Removed All Acrylic For Me

**Comment by jiangzhenjerry** [comment]
Wanted to report that this issue happens to me as well. I have `"useAcrylic": true` under `defaults`, and `"useAcrylicInTabRow": true,` as well. The outcome is that the acrylic effect is there, but the tab row is significant brighter/whiter than previous versions. 

Also, setting `"theme": "dark"` doesn't change the tab row color - still the same bright white.

I'm using a laptop with Intel CPU/GPU and NVIDIA GPU. By default It's Intel GPU to render.

**Comment by Mukul1127** [comment]
Maybe It's NVIDIA Issue

**Comment by jiangzhenjerry** [comment]
> Maybe It's NVIDIA Issue

No, it's not. I already clarified that it is using Intel GPU to render. Plus, this problem doesn't exist in any previous versions - nor in any other apps using acrylic - on my computer.

Don't say it is an Intel issue then - I hope this thread doesn't go from blaming AMD to blaming NVIDIA to blaming Intel... That would be a little excessive ;P

**Comment by Mukul1127** [comment]
Sorry I Didn't see that

**Comment by Mukul1127** [comment]
Drop-Down Acrylic Works

**Comment by Mukul1127** [comment]
Title Bar Acrylic Actually Still Works, Just Very White

**Comment by zadjii-msft** [comment]
Can I get a screenshot? I wanna know exactly what I'm looking for here. Light theme + acrylic titlebar does seem... off... in Windows 10, but I'm not sure that's what you're reporting here.

**Comment by zadjii-msft** [comment]
That's exactly the kind of thing I'm looking for. That's great, thanks!

**Comment by Mukul1127** [comment]
> 

I have The Exact Same Thing Important Parts, Here is An Image Also:
![Screenshot 2022-02-07 192509](https://user-images.githubusercontent.com/71291707/152894685-dc26cab0-feeb-41db-9b91-51d1df710f7a.png)

**Comment by Mukul1127** [comment]
Also Even If The Thing behind The Terminal Is Black, It Is Still White.

**Comment by zadjii** [comment]
system: dark. terminal: dark, 1.12:
![image](https://user-images.githubusercontent.com/534075/153203493-aa1d3024-a6ff-40b9-8556-537667ffdd71.png)
![image](https://user-images.githubusercontent.com/534075/153203635-11a86e6d-c24d-48e2-9e74-7842e726c5d8.png)

system: dark. terminal: dark, 1.13:
![image](https://user-images.githubusercontent.com/534075/153203564-bca55fbd-6e96-4395-bc1d-d82b162a9b28.png)


wait, how did that happen?

I was just messing around trying to screenshots on Win10, and that first 1.12 SS looks like it has the titlebar partially visible through the acrylic or something. I didn't actually seen that but clearly the SS grabbed it. 

Maybe that's what's affecting this. Maybe there's a weird DWM interaction between the system caption buttons. the transparent BG, and HostBackdropAcrylic

**Comment by zadjii** [comment]
OS: light, terminal: system. 1.12
![image](https://user-images.githubusercontent.com/534075/153205461-58cd9a93-a3d9-4378-a494-2e743d50f03f.png)

OS: light, terminal: system. 1.13
![image](https://user-images.githubusercontent.com/534075/153205487-62b5c6b8-7202-44a5-a0ae-fa954356d9de.png)


OS: light, terminal dark 1.12
![image](https://user-images.githubusercontent.com/534075/153205654-6938185b-2d2b-41e4-9e22-6d4a6bb6ad51.png)


OS: light, terminal dark 1.13
![image](https://user-images.githubusercontent.com/534075/153205623-74c5db02-786c-4af0-824f-a23e604ba392.png)


That's what it is. 🤦‍♂️

**Comment by zadjii-msft** [comment]
Regressed here: #12326 specifically: https://github.com/microsoft/terminal/pull/12326/files#diff-42b8e1723f803ad4ede2b8ca7547b6d2f9e3c12dd764a30e0789f379641e3b19


https://github.com/microsoft/microsoft-ui-xaml/blob/9052972906c8a0a1b6cb5d5c61b27d6d27cd7f11/dev/CommonStyles/Common_themeresources.xaml#L47

```xaml
<SolidColorBrush x:Key="ApplicationPageBackgroundThemeBrush" Color="{ThemeResource SystemColorWindowColor}" />
```

So we're clearly evaluating `ApplicationPageBackgroundThemeBrush` the same as the OS's `ApplicationPageBackgroundThemeBrush`, not the actual theme at runtime. This is gonna technically be the same thing as #3917, but way more impactful

**Comment by zadjii-msft** [comment]
I wonder if this is because we're doing the titlebar acylic brush creation manually. We're creating the acrylic brush based on the color that's already in `TabViewBackground`, which at this time is probably the light version of the resource - we haven't set up the `RequestedTheme` yet. 

The dumb solution would be to set `RequestedTheme` earlier, but that won't fix the hot-reload scenario here. We could theoretically manually re-resolve `ApplicationPageBackgroundThemeBrush` on a hot-reload, but that seems bodgy.

I don't think that would fix the pane brush either.

**Comment by zadjii-msft** [comment]
Note to self: `TitlebarControl`'s BG is also set to TabViewBackground. That may also have something to do with it. I doubt we can just bind the titlebar's BG to the content's BG

**Comment by zadjii-msft** [comment]
Hokay, I've spent more than enough time on this for now.

Something weird is definitely going on here. I'd like to have a better solution, but I don't. [`dev/migrie/b/12356-attempt-2`](https://github.com/microsoft/terminal/compare/dev/migrie/b/12356-attempt-2) has the latest state of my research into this. Creating a string literal xaml brush, parsing it, inserting it into the page's resources, then examining that brush's color at runtime worked, but _only_ after a call to `Dispatcher.RunAsync`, so that it happened on a subsequent pass, after the `Create` call finished. That's mental. 

I'm going with the hacky fix, which is in 82dd0e7c4. Just manually set the brush color for now. This is BODGY but hey, it's a good enough hotfix.

**Comment by Mukul1127** [comment]
> Hokay, I've spent more than enough time on this for now.
> 
> Something weird is definitely going on here. I'd like to have a better solution, but I don't. [`dev/migrie/b/12356-attempt-2`](https://github.com/microsoft/terminal/compare/dev/migrie/b/12356-attempt-2) has the latest state of my research into this. Creating a string literal xaml brush, parsing it, inserting it into the page's resources, then examining that brush's color at runtime worked, but _only_ after a call to `Dispatcher.RunAsync`, so that it happened on a subsequent pass, after the `Create` call finished. That's mental.
> 
> I'm going with the hacky fix, which is in [82dd0e7](https://github.com/microsoft/terminal/commit/82dd0e7c43489e917fe22d4b40063b7de6bbfb4d). Just manually set the brush color for now. This is BODGY but hey, it's a good enough hotfix.

How Do I Use This Fix?

**Comment by zadjii-msft** [comment]
The fix was merged in #12460, which we haven't shipped yet. When a build with that fix is released, the bot will come through and ping this thread.

For now, a workaround would be to set your OS theme to match the Terminal theme. (e.g. set the OS to Dark theme)
